apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlTests.cs

```diff
@@ -535,42 +535,54 @@ public void PartitionedUpdate_Medium()
             }
         }
 
-        [Fact(Skip = "This test is skipped until DML Returning changes are done.")]
+        // With DML's return clause ExecuteReaderAsync should return a reader containing affected rows data.
+        [Fact]
+        [Trait(Constants.SupportedOnEmulator, Constants.No)]
         public async Task DMLReturn_ExecuteReader_Read()
         {
             string key = _fixture.CreateTestRows();
-            var actualValues = new List<int>();
+            var actualReturnedValues = new List<int>();
 
             using var connection = _fixture.GetConnection();
             await connection.OpenAsync();
-            RetryHelpers.ExecuteWithRetry(async () =>
+            string dml = $"UPDATE {_fixture.TableName} SET Value = OriginalValue + 1 WHERE UpdateMe AND Key=@key Then Return Value";
+            var command = connection.CreateDmlCommand(dml);
+            command.Parameters.Add("key", SpannerDbType.String, key);
+            using var reader = await command.ExecuteReaderAsync();
+            while (await reader.ReadAsync())
             {
-                string dml = $"UPDATE {_fixture.TableName} SET Value = OriginalValue + 1 WHERE UpdateMe AND Key=@key Then Return Value";
-                var command = connection.CreateDmlCommand(dml);
-                using var reader = await command.ExecuteReaderAsync();
-                while (await reader.ReadAsync())
-                {
-                    actualValues.Add(reader.GetFieldValue<int>(reader.GetOrdinal("Value")));
-                }
-            });
-            var expectedValues = new List<int> { 2, 5 };
-            Assert.Equal(expectedValues, actualValues);
+                actualReturnedValues.Add(reader.GetFieldValue<int>(reader.GetOrdinal("Value")));
+            }
+            // TO Assert that DML Retrun command has returned expected values only.
+            var expectedReturnedValues = new List<int> { 2, 5 };
+            Assert.Equal(expectedReturnedValues, actualReturnedValues);
+
+            // To Assert that DML Return command has actually updated the table.
+            var actualDbValues = _fixture.FetchValues(key);
```

Comment:
Let's extract this code to another method, given that it's used for multiple tests.